generator/booking_dataset.hpp

```diff
@@ -0,0 +1,70 @@
+#pragma once
+
+#include "generator/osm_element.hpp"
+
+#include "boost/geometry.hpp"
+#include "boost/geometry/geometries/point.hpp"
+#include "boost/geometry/geometries/box.hpp"
+#include "boost/geometry/index/rtree.hpp"
+
+#include "std/function.hpp"
+#include "std/string.hpp"
+
+class BookingDataset
+{
+public:
+  struct BookingHotel
+  {
+    enum class Fields : size_t
```

Comment:
Do you really need such strong typing here, as you use this enum only with static_cast?